README.markdown

```diff
@@ -273,7 +273,7 @@ Very much like [more_set_headers](#more_set_headers) except that it operates on
 
 Note that using the `-t` option in this directive means filtering by the `Content-Type` *request* header, rather than the response header.
 
-Behind the scene, use of this directive and its friend [more_clear_input_headers](#more_clear_input_headers) will (lazily) register a `rewrite phase` handler that modifies `r->headers_in` the way you specify. Note that it always run at the *end* of the `rewrite` so that it runs *after* the standard [rewrite module](http://nginx.org/en/docs/http/ngx_http_rewrite_module.html) and works in subrequests as well.
+Behind the scene, use of this directive and its friend [more_clear_input_headers](#more_clear_input_headers) will (lazily) register a `rewrite phase` handler that modifies `r->headers_in` the way you specify. Note that it always run at the *end* of the `rewrite` so that it runs *after* the standard [rewrite module](http://nginx.org/en/docs/http/ngx_http_rewrite_module.html) and works in subrequests as well.  In addition, this means that set or cleared headers will not yet have been modified in directives that run *before* this phase (such as `return` and `rewrite`). Use of corresponding variables in these types of directives will still show their original values as they are not yet cleared.
```

Comment:
I raised my original issue because the issue was confusing for someone without detailed knowledge of nginx's request phases (I'd imagine that would be most people).  Put simply, it wasn't clear why those variables were still set until I learned the key fact that certain directives run before headers-more. 

My suggestions here highlight two of these directives (return and rewrite) where the variables won't have been cleared.  Those were the two I'd been using (and confused over) -- so felt highlighting them would help a future reader hitting the same issue.

Either which way, I understand it now having gone through it and these comments will turn up in web searches to help anyone who might be confused about this later.